github.ghactivity.php

```diff
@@ -289,4 +290,47 @@ private function get_repo_activity( $repo_name = '' ) {
 
 		return $this->get_github_data( $query_url );
 	}
+
+	/**
+	 * Get the list of repo label names from GH API
+	 *
+	 * @since 2.0.0
+	 *
+	 * @param string $repo_name Name of the repo we are interested in.
+	 *
+	 * @return array Array of repo label names.
+	 */
+	public function get_repo_label_names( $repo_name ) {
+		if ( empty( $repo_name ) ) {
+			// Fallback.
+			return array();
+		}
+
+		$page       = 1;
+		$all_labels = array();
+		$query_url  = sprintf(
+			'https://api.github.com/repos/%1$s/labels?access_token=%2$s&per_page=100',
+			$repo_name,
+			$this->token
+		);
+
+		// Fetch API until empty array will be returned.
+		do {
+			$paged_query_url = $query_url . '&page=' . $page;
+			$body            = $this->get_github_data( $paged_query_url );
+			$all_labels      = array_merge( $all_labels, $body );
```

Comment:
Is this enough to ensure that we do not store duplicate labels?

Reply:
I think so. It relies on data returned from GH, which shouldn't include duplicate labels. Do you think it worth to clean up possible duplicates?

Reply:
I have not tested, so you know best, I trust you with that
!